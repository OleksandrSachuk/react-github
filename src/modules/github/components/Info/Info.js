import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

const propTypes = {
  info: PropTypes.object,
};

const Info = ({ info }) => (
  <Fragment>
    { info.id
      ? <div>
        <div>Github user information:</div>
        <Table>
          <TableBody>

            <TableRow key={ info.name }>
              <TableCell component="th" scope="row">
                Name
              </TableCell>
              <TableCell>{ info.name }</TableCell>
            </TableRow>

            <TableRow key={ info.email }>
              <TableCell component="th" scope="row">
                Email
              </TableCell>
              <TableCell>{ info.email }</TableCell>
            </TableRow>

            <TableRow key={ info.location }>
              <TableCell component="th" scope="row">
                Location
              </TableCell>
              <TableCell>{ info.location }</TableCell>
            </TableRow>

            <TableRow key={ info.followers }>
              <TableCell component="th" scope="row">
                Number of follower
              </TableCell>
              <TableCell>{ info.followers }</TableCell>
            </TableRow>

          </TableBody>
        </Table>

      </div>
      : null
    }
  </Fragment>
);

Info.propTypes = propTypes;

export default Info;
