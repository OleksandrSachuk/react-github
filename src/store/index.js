import { createStore, applyMiddleware } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import rootEpic from '../modules/github/epics';

import rootReducer from '../modules/github/reducers';

const epicMiddleware = createEpicMiddleware();

const store = createStore(
  rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  applyMiddleware(epicMiddleware),
);

epicMiddleware.run(rootEpic);

export default store;