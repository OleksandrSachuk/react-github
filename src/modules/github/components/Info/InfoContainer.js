import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Info from './Info';

const mapStateToProps = (state) => ({
  info: state.user.info,
});

const mapDispatchToProps = (dispatch) => bindActionCreators({}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Info);
