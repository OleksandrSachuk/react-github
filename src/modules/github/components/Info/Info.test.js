import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Info from './';
import store from '../../../../store';

it('renders Info without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Info store={ store }/>, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('Info snapshot test', () => {

  const component = renderer.create(
      <Info store={ store }/>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
