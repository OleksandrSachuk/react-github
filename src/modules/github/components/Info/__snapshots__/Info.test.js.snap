// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info snapshot test 1`] = `null`;
